### PR TITLE
Migrate docker rm tests to api tests

### PR DESCRIPTION
**Pull request by yongtang**
This fix migrates docker rm test in integration-cli
to api tests.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: integration/container/remove_test.go [hunk 1]
```diff
@@ -111,3 +117,61 @@ func TestRemoveInvalidContainer(t *testing.T) {
...
+	resp, _, err := client.ImageInspectWithRaw(ctx, imageID1)
+	require.NoError(t, err)
+	assert.Equal(t, resp.ID, imageID1)
+
+	// check if the second image has been deleted
+	_, _, err = client.ImageInspectWithRaw(ctx, imageID2)
+	testutil.ErrorContains(t, err, "No such image:")
+}
+
+func buildImageWithDockerfile(t *testing.T, ctx context.Context, client client.APIClient, tag string, dockerfile string) string { // nolint: golint
```
Comment (AkihiroSuda) [inline integration/container/remove_test.go]: Maybe we can expose this function to other packages, but not needed right now

**Review by AkihiroSuda — APPROVED** [review]

File: integration/container/remove_test.go [hunk 2]
```diff
@@ -111,3 +117,61 @@ func TestRemoveInvalidContainer(t *testing.T) {
...
+
+	dockerfile1 := `FROM busybox:latest
+	ENTRYPOINT ["true"]`
+
+	dockerfile2 := `FROM busybox:latest
+	ENTRYPOINT ["true"]
+	MAINTAINER Integration Tests`
+
+	// build first dockerfile
+	imageID1 := buildImageWithDockerfile(t, ctx, client, img, dockerfile1)
```
Comment (dnephin) [inline integration/container/remove_test.go]: I think we should avoid using `build` in tests outside of the image suite.

1. It's going to be slower to build something that to `load`
2. It means that multiple test suites require build. If we want to provide build as a separate service this unnecessary coupling is going to make it a lot more work.

Can we setup this image structure using `image load` api?  If not,  we should make this a unit test.

**Review by dnephin — CHANGES_REQUESTED** [review]

**Comment by yongtang** [comment]
@dnephin The PR has been updated to use commit for a small image change (vs. Dockerfile) instead. Please take a look.

**Review by dnephin — COMMENTED** [review]
Thanks for making this change.

Looking at the test in more detail now, it seems like it is testing "image remove" not container remove. Is that right?

Should this move to `integration/image/remove_test.go` ?

**Comment by yongtang** [comment]
@dnephin Thanks. Yes it might be more relevant to images. The PR has been updated. Please take a look.

**Review by dnephin — APPROVED** [review]
LGTM

**Pull request merged**